_overviews/scala3-scaladoc/snippet-compiler.md

```diff
@@ -0,0 +1,223 @@
+
+To do this, you can add a flag directly to the snippet:
+
+````
+```scala sc:nocompile
+// snippet
+```
+````
+
+Sometimes compilation failure is an intended behavior. For that case, we exposed a flag `fail` that introduces one of our features: [Assert compilation errors](#assert-compilation-errors).
+
+````
+```scala sc:fail
+List(1,2,3).toMap
+```
+````
+
+For more thorough explanation and more sophisticated configuration e.g. path-based flag settings see [Advanced configuration](#advanced-configuration) section.
+
+## Features overview
+
+### Assert compilation errors
+
+Scala with its rich type system allows us to ensure almost complete correctness of code in compile-time. Sometimes, documentation should mention case where code should not compile or authors wants to provide ways to recover from certain compilation errors.
+
+Example:
+
+```scala
+List(1,2,3).toMap
+```
+
+results in:
+
+```nohighlight
+
+At 18:21:
+  List(1,2,3).toMap
+Error: Cannot prove that Int <:< (K, V)
+
+where:    K is a type variable with constraint 
+          V is a type variable with constraint 
+.
+```
+
+Examples presenting code that fails at compile-time are sometimes very important. For example you can show how the library is secured against incorrect code. The other use case is to present common mistakes and how to solve them. Taking that into account, we decided to provide a functionality to check if the marked code snippets don’t compile.
+
+For snippet that compiles without error:
+
+```scala
+List((1,2), (2,3)).toMap
+```
+
+results in:
+
+```nohighlight
+
+In static site (./docs/docs/index.md):
+Error: Snippet should not compile but compiled succesfully
+```
+
+For snippet that fails to compile:
+
+```scala
+List(1,2,3).toMap
+```
+
+Passes the snippet validation and shows expected compilation errors in documentation.
+
+![]({{ site.baseurl }}/resources/images/scala3/scaladoc/assert-compilation-errors.gif)
+
+### Context
+
+Our goal is to make snippets in docstrings behave as much as possible as it is defined within a given scope (in certain package, inside a class). For us it brings a natural feel to snippets. To achieve this, we implemented a wrapping mechanism that provides a context for each snippet.
+
+Let's assume that we want to document method `slice` in a `collection.List`. 
+
+We want to explain how it work by comparing it to combination of `drop` and `take` method so using snippet like:
+```scala
+slice(2,5) == drop(2).take(3)
+```
+is one of the first thing that comes to mind. As you probably have guessed, this will not compile without a **context** feature.
+
+Besides our main goal, it reduces the boilerplate of a snippet, because you don’t need to import members of the same package and instantiate documented class.
+
+For people that are curious how our context mechanism works, the snippet after preprocessing will look like this:
```

Comment:
I feel like something is missing, how do we use the “context” feature?

Reply:
Added a part that explains that context feature is automatically turned on

Reply:
```suggestion
For people that are curious on how our context mechanism works, the snippet after preprocessing looks like this:
```